source/mir/random/discrete.d

```diff
@@ -67,8 +67,9 @@ struct Discrete(T)
     */
     this(const(T)[] cdPoints)
     {
+        import std.range : assumeSorted;
         _cdPoints = cdPoints;
-        r = typeof(r)(_cdPoints);
+        _cdSortedRange = _cdPoints.assumeSorted!"a <= b";
```

Comment:
verify sorted occurs after the SortedRange constructor has been called:
https://github.com/dlang/phobos/blob/master/std/range/package.d#L7919

[`assumeSorted`] (https://github.com/dlang/phobos/blob/master/std/range/package.d#L8514) is:

```d
auto assumeSorted(alias pred = "a < b", R)(R r)
{
    return SortedRange!(Unqual!R, pred)(r);
}
```

so imho I would keep the direct constructor call..